### DISCUSSION TITLE
nodes to generate keystore for OIDC-provider

### DISCUSSION THREAD

**Discussion by piyushSinghalDemo** [panva/jose#486] [Q&A]
hi , i am upgrading jose package For this in my oidc-provider needs to generate JWKS keystore previously there was option to generate like
const keys = [
      jose.JWK.generateSync('RSA', 2048, {
        kid: 'sig-rs-0',
        use: 'sig',
      }),
      jose.JWK.generateSync('RSA', 2048, {
        kid: 'enc-rs-0',
        use: 'enc',
      })
.....
.....
]
*******but now in jose v4 i am trying to generate like
const keys = [
        await jose.generateKeyPair('PS256',{ extractable: true }),
        await jose.generateKeyPair('PS256',{ extractable: true }),
        await jose.generateKeyPair('ES384',{ extractable: true }),
        .................
    ];

    let keystoreData=[];
    for(let i=0;i<keys.length;i++){
      const exportedKey = await jose.exportJWK(keys[i].privateKey);      
      keystoreData.push(exportedKey);
    }
is there any other related method so that i can get kid & use in key store so that it will work with oidc-provider jwks config.

**Comment by panva** [comment]
`kid` is an arbitrary unique string, feel free use a random uuid, or if you want a deterministic unique identifier you may use a [JWK's thumbprint](https://github.com/panva/jose/blob/main/docs/functions/jwk_thumbprint.calculateJwkThumbprint.md#readme).

`use` is the key's usage determination, `sig` for signatures, `enc` for encryption.

As these are all just plain objects I believe assigning these properties should need no explanation.
